hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/HealthyPipelineSafeModeRule.java

```diff
@@ -204,15 +206,16 @@ public String getStatusText() {
         "healthy Ratis/THREE pipelines (=%d) >= healthyPipelineThresholdCount" +
             " (=%d)", getCurrentHealthyPipelineCount(),
         getHealthyPipelineThresholdCount());
-    Set<PipelineID> samplePipelines = pipelineIDSet.stream()
-        .filter(element -> !processedPipelineIDs.contains(element))
-        .limit(SAMPLE_PIPELINE_DISPLAY_LIMIT)
-        .collect(Collectors.toSet());
-
-    if (!samplePipelines.isEmpty()) {
-      String samplePipelineText =
-          "Sample pipelines not satisfying the criteria : " + samplePipelines;
-      status = status.concat("/n").concat(samplePipelineText);
+    synchronized (unProcessedPipelineSet) {
```

Comment:
`synchronized (unProcessedPipelineSet)` does not protect access, because we sync on other object (`this`) when changing the set.  I think the simplest fix is to make this method `synchronized`.